Documentation/proposals/202501-configuration-object-status-subresource.md

```diff
@@ -0,0 +1,193 @@
+## Status Subresource For Config-Based Resources
+
+* **Owners:**
+  * [yp969803](https://github.com/yp969803)
+* **Status:**
+  * `Accepted`
+* **Related Tickets:**
+  * [#3385](https://github.com/prometheus-operator/prometheus-operator/issues/3335)
+* **Other docs:**
+  * NA
+
+> This proposal describes how we will extend the Prometheus operator configuration Custom
+> Resource Definitions (CRDs) with a Status subresource field.
+
+## Why
+
+This will enhance observability by allowing users to verify whether their configurations have been successfully applied to Workload.
+
+## Pitfalls of the current solution
+
+Currently, the status subresource is only implemented for workload resources. This makes it difficult to determine which workloads are using which configurations—especially since a single workload can reference multiple configurations. Additionally, there is no straightforward way to observe the reconciliation status of configuration resources. While Kubernetes events are available when a configuration is rejected by a workload, they are not sufficient for ongoing visibility or troubleshooting.
+
+## Goals
+
+* Define the structure of the status subresource for the custom resource definitions
+  * `ServiceMonitor`
+  * `PodMonitor`
+  * `ScrapeConfig`
+  * `Probes`
+  * `PrometheusRule`
+  * `AlertmanagerConfig`
+* Information about number of Up/Down targets in status subresource for PodMonitor, ServiceMonitor, Probes and ScrapeConfig.
+* Define how the operator would reconcile the status subresource.
+* Feature gate to help the user to enable/disable the feature.
+
+### Audience
```

Comment:
This section should cover who are the intended audience of this Proposal. In this case, it would be 
Users of Prometheus-Operator
Maintainers and Contributors of Prometheus-Operator
Consumers of Prometheus-Operator CRDs (there are other OSS that consume our CRDs like ServiceMon and PodMon)